camera_system/camera_coordinate_calibrator.py

```diff
@@ -0,0 +1,156 @@
+"""カメラ画像の座標取得するためのモジュール.
+
+カメラ画像を表示し、クリックした座標を保持する
+@author miyashita64 mutotaka0426
+@note 参考: https://techacademy.jp/magazine/51035
+@note 参考:
+    - [【Python】Tkinterによる画像表示をわかりやすく解説](https://onl.bz/5Bwmt1b)
+    - [Tkinterで作成したウインドウにOpenCV-Pythonの画像を表示](https://onl.bz/sTkdsGb)
+    - [Tkinter: イベントを検出する（クリック・キー入力・マウス移動）](https://onl.bz/7sMdTKw)
+    - [Python Tkinter Canvasについて](https://onl.bz/mWGj8f9)
+    - [【Python/tkinter】ウィジェットの配置(pack)](https://onl.bz/etTHi5T)
+"""
+
+import cv2
+import tkinter as tk
+from PIL import Image, ImageTk
+from typing import List, Tuple
+
+
+class CameraCoordinateCalibrator:
+    """カメラ画像から座標を取得するクラス."""
+
+    def __init__(self, img: cv2.Mat) -> None:
+        """CameraCoordinateCalibratorのコンストラクタ.
+
+        Args:
+            img (cv2.Mat): 画像データ
+        """
+        # メンバを初期化する
+        self.__block_point = []  # ブロック置き場の座標リスト
+        self.__base_circle = []  # ベースサークルの座標リスト
+        self.__end_point = []  # 端点サークルの座標リスト
+        self.__img = img
+
+    def show_window(self) -> None:
+        """画像取得ツールを起動する関数."""
+        # 画像情報を取得する
+        img_height = self.__img.shape[0]  # 画像の高さ
+        img_width = self.__img.shape[1]  # 画像の横幅
+
+        # ウィンドウを定義する
+        self.__window = tk.Tk()
+        self.__window.title("Camera Coordinate Calibrator")
+        # ウィンドウサイズ: (画像の横幅 + UIの横幅) x 画像の高さ
+        self.__window.geometry("%dx%d" % (img_width+200, img_height))
+
+        # 取得状況を表示するMessageを定義する
+        self.__message = tk.Message(self.__window, text="", font=("", 10), bg="#ddd", aspect=500)
+        # Messageを配置する
+        self.__message.place(x=img_width+10, y=80, width=180)
+
+        # OpenCVで取得した画像を変換する
+        img_rgb = cv2.cvtColor(self.__img, cv2.COLOR_BGR2RGB)  # imreadはBGRなのでRGBに変換
+        img_pil = Image.fromarray(img_rgb)    # RGBからPILフォーマットへ変換
+        img_tk = ImageTk.PhotoImage(img_pil)  # ImageTkフォーマットへ変換
+
+        # ウィジェットを定義する
+        # 画像を表示するCanvasを定義する
+        canvas = tk.Canvas(self.__window, width=img_width, height=img_height)
+        # コールバック関数を指定する ("<Button-1>"は左クリックボタン)
+        canvas.bind("<Button-1>", self.__set_coordinate)
+        # Canvasを配置する
+        canvas.pack(side=tk.LEFT)
+        # Canvasに画像を設置する
+        canvas.create_image(0, 0, image=img_tk, anchor=tk.NW)
+
+        # リセット用のButtonを定義する
+        reset_button = tk.Button(self.__window, text="Reset")
+        # コールバック関数を指定する ("<Button-1>"は左クリックボタン)
+        reset_button.bind("<Button-1>", self.__reset_coordinate)
+        # リセットボタンを配置する
+        reset_button.place(x=img_width+10, y=20, width=180, height=50)
+
+        # ウィンドウを表示する
+        self.__window.mainloop()
+
+    def __set_coordinate(self, event) -> None:
+        """マウス操作で取得した座標を各座標リストにセットするコールバック関数.
+
+        Args:
+            event: イベント

```

Comment:
引数名以上の情報を説明には付けたほうが良い気がしてきた（今更すみません...）
```suggestion
            event: マウスイベント
```